docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,196 @@
+
+In the future, we could migrate the current standalone openshift-install implementation to an OpenShiftInstallStrategy, which continues to use ClusterProvision for it's retry mechanism. To do this we would allow a parallel path to specify where you cannot use both installStrategyRef and provisioning sections together. Provisioning then becomes deprecated and after some period of time, could be removed.
+
+We could add an additional strategy which offers InstallConfig generation, and could be used by users who are ok with a simpler interface and don't want to specify an install config secret.
+
+HypershiftInstallStrategy would be another likely future implementation.
+
+### Examples
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: AgentInstallStrategy
+metadata:
+  name: mycluster
+  namespace: hive
+spec:
+  stage: RetryProvisioning
+  clusterDeploymentRef:
+    name: mycluster
+  imageSetRef:
+    openshift-4.6
+  clusterMetadata:
+    # set as soon as known even if install is not completed:
+    adminKubeconfigSecretRef:
+      name: mycluster-3-2sw49-admin-kubeconfig
+    adminPasswordSecretRef:
+      name: mycluster-3-2sw49-admin-password
+    clusterID: 48dec7cf-0c4b-44c6-b840-5c8a6ec93aa6
+    infraID: mycluster-4n2t5
+
+  # agent specific
+  networking:
+    machineNetwork: ...
+  sshPublicKey: ...
+  provisionRequirements:
+    controlPlaneAgents: 3
+    workerAgents: 5
+  controlPlane:
+    hyperthreading: Disabled
+  compute:
+    hyperthreading: Disabled
+
+  status:
+    conditions:
+    - lastProbeTime: "2021-04-15T15:50:59Z"
+      lastTransitionTime: "2021-04-15T15:50:59Z"
+      message: Provision attempt 2 of 3 underway.
+      reason: ProvisionRetryUnderway
+      status: "False"
+      type: Ready
+    - lastProbeTime: "2021-04-15T15:50:59Z"
+      lastTransitionTime: "2021-04-15T15:50:59Z"
+      message: All requirements met.
+      reason: AllRequirementsMet
+      status: "True"
+      type: RequirementsMet
+    - lastProbeTime: "2021-04-15T15:50:59Z"
+      lastTransitionTime: "2021-04-15T15:50:59Z"
+      message: Quota exceeded
+      reason: SomeQuotaExceeded
+      status: "True"
+      type: Failed
+    - lastProbeTime: "2021-04-15T15:50:59Z"
+      lastTransitionTime: "2021-04-15T15:50:59Z"
+      message: Attempt x 2 of 3 underway.
+      reason: ProvisionUnderway
+      status: "True"
+      type: Retrying
+
+    - lastProbeTime: "2021-04-15T15:50:59Z"
+      lastTransitionTime: "2021-04-15T15:50:59Z"
+      message: Some agent specific condition.
+      reason: AgentReason
+      status: "True"
+      type: AgentSpecificCondition
+```
+
+
+### Risks and Mitigations
+
+  * Can Hive controllers watch a dynamic set of types to reconcile when an install strategy changes? Can we watch unstructured?
+  * Is InstallStrategy the naming we're happy with? Consider ClusterProvision off the table given it's use already for another purpose.
```

Comment:
i would like the contract to be called ClusterInstall , installstrategy doesn't should like a good name for this interface

Reply:
The whole time I've been reading this, the word "Strategy" felt like the wrong noun, and I figured I'd wait until I got to the end to bring it up. Having arrived, I'm happy to see that I'm not alone. :)

If I could wave a magic wand, I might call this new resource `ClusterDeployment` and call the other thing `Cluster`. But those ships have sailed.

`ClusterInstall` seems like a reasonable option.

Reply:
We've always avoided install in favor of provision/deprovision, but I guess with those terms out of the running, ClusterInstall might work.